### PR TITLE
Arm64: Add AddAcross Codegen and Tests.

### PR DESCRIPTION
**Pull request by TamarChristinaArm**
Hi All,

This completes the `Addv` implementation.

As discussed I've changed the intrinsics to return a Vector so the RA is able to pic the correct register file.

The disassembly is a bit misleading though:

```
G_M48057_IG04:
        3DC00BB0          ldr     q16, [fp,#32]
        4EB1BA10          addv    v16.4s, v16.4s
        3D8007B0          str     q16, [fp,#16] // [V02 tmp1]
        3DC007B0          ldr     q16, [fp,#16] // [V02 tmp1]
        4EB01E00          mov     v0.16b, v16.16b
```

Would it make sense to try to change the disassembler to emit

```
addv    s16, v16.4s
```
here?

/cc @CarolEidt @tannergooding @echesakovMSFT

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs [hunk 1]
```diff
@@ -50,61 +50,61 @@ public new abstract class Arm64 : ArmBase.Arm64
...
             /// </summary>
-            public static int AddAcross(Vector128<int> value) { throw new PlatformNotSupportedException(); }
+            public static Vector128<int> AddAcross(Vector128<int> value) { throw new PlatformNotSupportedException(); }
 
             /// <summary>
             /// int8_t vaddvq_s8(int8x16_t a)
             ///   A64: ADDV Bd, Vn.16B
             /// </summary>
-            public static sbyte AddAcross(Vector128<sbyte> value) { throw new PlatformNotSupportedException(); }
+            public static Vector128<byte> AddAcross(Vector128<sbyte> value) { throw new PlatformNotSupportedException(); }
```
Comment (echesakovMSFT) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs]: Shouldn't the function return type be `Vector128<sbyte>`?

**Review by CarolEidt — COMMENTED** [review]
Aside from the one type mismatch @echesakovMSFT pointed out, this LGTM

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs (hunk 1 above)
Comment (TamarChristinaArm) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs]: Thanks!

**Review by echesakov — APPROVED** [review]
Looks Good

**Comment by echesakovMSFT** [comment]
win-arm failures are not related to these changes and tracked by https://github.com/dotnet/runtime/issues/1097

**Pull request merged**